crates/store/re_chunk_store/src/query.rs

```diff
@@ -413,3 +560,201 @@ impl ChunkStore {
+        component_name: ComponentName,
+    ) -> usize {
+        re_tracing::profile_function!();
+
+        self.query_id.fetch_add(1, Ordering::Relaxed);
+
+        if let Some(static_chunk) = self
+            .static_chunk_ids_per_entity
+            .get(entity_path)
+            .and_then(|static_chunks_per_component| {
+                static_chunks_per_component.get(&component_name)
+            })
+            .and_then(|chunk_id| self.chunks_per_chunk_id.get(chunk_id))
+        {
+            static_chunk
+                .num_events_for_component(component_name)
+                .unwrap_or(0)
+        } else {
+            0
+        }
+    }
+
+    /// Returns the number of times a temporal component was logged to an entity path on a specific timeline.
+    ///
+    /// This ignores static data.
+    pub fn num_events_on_timeline_for_temporal_component(
+        &self,
+        timeline: &Timeline,
+        entity_path: &EntityPath,
+        component_name: ComponentName,
+    ) -> usize {
+        re_tracing::profile_function!();
+
+        self.query_id.fetch_add(1, Ordering::Relaxed);
+
+        let Some(temporal_chunk_ids_per_timeline) = self
+            .temporal_chunk_ids_per_entity_per_component
+            .get(entity_path)
+        else {
+            return 0; // no events logged for the entity path
+        };
+
+        let Some(temporal_chunk_ids_per_component) = temporal_chunk_ids_per_timeline.get(timeline)
+        else {
+            return 0; // no events logged on this timeline
+        };
+
+        let Some(chunk_ids) = temporal_chunk_ids_per_component.get(&component_name) else {
+            return 0; // no events logged for the component on this timeline
+        };
+
+        let mut num_events = 0;
+        for chunk in chunk_ids
+            .per_start_time
+            .values()
+            .flat_map(|ids| ids.iter().filter_map(|id| self.chunks_per_chunk_id.get(id)))
+        {
+            num_events += chunk.num_events_for_component(component_name).unwrap_or(0);
+        }
+
+        num_events
+    }
+
+    /// Returns the size of the entity on a specific timeline.
+    ///
+    /// This includes static data.
+    ///
+    /// ⚠ This does not return the _total_ size of the entity and all its children!
+    /// For that, use `entity_db.size_of_subtree_on_timeline`.
+    pub fn size_of_entity_on_timeline(
+        &self,
+        timeline: &Timeline,
+        entity_path: &EntityPath,
+    ) -> usize {
+        re_tracing::profile_function!();
+
+        self.query_id.fetch_add(1, Ordering::Relaxed);
+
+        let mut total_size = 0;
+
+        if let Some(static_chunks_per_component) = self.static_chunk_ids_per_entity.get(entity_path)
+        {
+            for chunk in static_chunks_per_component
+                .values()
+                .filter_map(|id| self.chunks_per_chunk_id.get(id))
+            {
+                total_size += Chunk::total_size_bytes(chunk) as usize;
```

Comment:
I see you've learned from my mistakes :D

Reply:
It is _really_ surprising that `Arc` implements `heap_size_bytes`, but returns `0` for it... 😅